tests/unit-tests/test-class-lesson.php

```diff
@@ -101,13 +100,110 @@ public function testIsPreRequisiteComplete() {
 
 	}
 
+	public function testGetCourseId() {
+		$this->assertTrue(
+			method_exists( 'WooThemes_Sensei_Lesson', 'get_course_id' ),
+			'The lesson class method `get_course_id` does not exist '
+		);
+		$course_ids             = $this->factory->course->create_many( 3 );
+		$lesson_ids             = $this->factory->lesson->create_many( 9 );
+		$lesson_id_to_course_id = array();
+		foreach ( $lesson_ids as $lesson_id ) {
+			$course_index                         = array_rand( $course_ids );
+			$course_id                            = $course_ids[ $course_index ];
+			$lesson_id_to_course_id[ $lesson_id ] = $course_id;
+			update_post_meta( $lesson_id, '_lesson_course', $course_id );
+		}
+		foreach ( $lesson_id_to_course_id as $lesson_id => $expected_course_id ) {
+			$course_id = Sensei()->lesson->get_course_id( $lesson_id );
+			$this->assertEquals(
+				$expected_course_id,
+				$course_id,
+				"Lesson with ID {$lesson_id} has course ID {$course_id}, expected {$expected_course_id}"
+			);
+		}
+	}
+
+	public function testGetCourseIds() {
+		$this->assertTrue(
+			method_exists( 'WooThemes_Sensei_Lesson', 'get_course_ids' ),
+			'The lesson class method `get_course_ids` does not exist '
+		);
+		$course_ids             = $this->factory->course->create_many( 3 );
+		$lesson_ids             = $this->factory->lesson->create_many( 9 );
+		$lesson_id_to_course_id = array();
+		foreach ( $lesson_ids as $lesson_id ) {
+			$course_index                         = array_rand( $course_ids );
+			$course_id                            = $course_ids[ $course_index ];
+			$lesson_id_to_course_id[ $lesson_id ] = $course_id;
+			update_post_meta( $lesson_id, '_lesson_course', $course_id );
+		}
+		$courses_id = Sensei()->lesson->get_course_ids( $lesson_ids );
+		foreach ( $lesson_id_to_course_id as $lesson_id => $expected_course_id ) {
+			$course_id            = $courses_id[ $lesson_id ];
+			$get_course_id_result = Sensei()->lesson->get_course_id( $lesson_id );
+			$this->assertEquals(
+				$expected_course_id,
+				$course_id,
+				"Lesson with ID {$lesson_id} has course ID {$course_id}, expected {$expected_course_id}"
+			);
+			$this->assertEquals(
```

Comment:
I think this test is the same as the previous one, just including this extra assertion, right? If that's correct, should we remove the previous test?

Reply:
Yep, it's mostly for testing the cache. That's why it looks so similar.

I thought about making it the same, but I wasn't 100% sure. I'll change it then..

Reply:
Considering that `testGetCourseId` and `testGetCourseIds` are different tests for different methods, I don't feel like removing any of these tests. I removed the test that verified against the cache for the `testGetCourseIds` method, tho, in [d0963e5](https://github.com/Automattic/sensei/pull/5294/commits/d0963e517d031e03a26b97966f612fa48188cfa8).